File: model/base/GameBase.ts
import Penalty from "../Penalty";

type GameIntimacy = "HIGH" | "LOW";
type GamePlayDirection = "LEFT" | "RIGHT" ;

interface IBaseGame {
    intimacy: GameIntimacy;
    recommendedPeopleRange: [number, number];
    direction: GamePlayDirection;
    description: string;
    name: string; 
}
 
/**
 * Game의 기본적인 기능을 구현하는 Base model
 * 이걸 확장시켜서 게임을 제작함
 */
export default abstract class GameBase {
    // 친밀도
    intimacy: GameIntimacy;
    // 권장 인원
    recommendedPeopleRange: [number, number];
    // 방향
    direction: GamePlayDirection;
    // 설명
    description: string;
    // 게임 이름
    name: string; 
    
    constructor({ intimacy, recommendedPeopleRange, direction, description, name }: IBaseGame) {
        this.intimacy = intimacy;
        this.recommendedPeopleRange = recommendedPeopleRange;
        this.direction = direction;
        this.description = description;
        this.name = name;
    }

    public abstract game(): void;

    public playPenalty() {
        new Penalty().randomPlayPenalty(this.intimacy);
    }

    public gameStart() {
        this.game();
    }

    public gameEnd() {
        
    }
    
}

File: model/GameBase.py
from abc import *
from Penalty import Penalty

# py-script는 핸들러와 연결만 할 예정.
# 웬만하면 모든 작업은 여기에서 이뤄질 수 있도록 함.
class GameBase(metaclass=ABCMeta):
    def __init__(self, intimacy, recommendedPeopleRange, direction, description, name):
        # 친밀도 
            # 0~N까지, 높을수록 친밀한 값
        self.intimacy = intimacy
        # 권장 인원
        self.recommendedPeopleRange = recommendedPeopleRange
        # 방향
        self.direction = direction
        
        # 설명
        self.description = description
        # 게임 이름
        self.name = name
        
    @abstractmethod
    def start_game(self):
        pass
    
    def end_game(self):
        pass
    
    def play_random_penalty(self):
        penalty = Penalty()
    
        return penalty.play_random_penalty()

    